polpo/lddmm/parallel_transport.py

```diff
@@ -0,0 +1,213 @@
+        "concentration_of_time_points": number_of_time_steps,
+        "number_of_time_points": number_of_time_steps + 1,
+        "use_rk2_for_flow": use_rk2_for_flow,
+        "use_rk2_for_shoot": use_rk2_for_shoot,
+        "output_dir": output_dir,
+        "write_adjoint_parameters": write_params,
+    }
+
+    template_specifications = {
+        "shape": {
+            "deformable_object_type": "landmark",
+            "kernel_type": kernel_type,
+            "kernel_width": kernel_width,
+            "kernel_device": kernel_device,
+            "noise_std": regularisation,
+            "filename": source,
+            "noise_variance_prior_scale_std": None,
+            "noise_variance_prior_normalized_dof": 0.01,
+        }
+    }
+
+    Deformetrica(output_dir, verbosity="INFO")
+    compute_shooting(
+        template_specifications,
+        initial_control_points=control_points,
+        external_forces=external_forces,
+        initial_momenta=momenta,
+        **deformation_parameters,
+    )
+
+    return time.gmtime()
+
+def estimate_parallel_transport(
+        source, 
+        target, 
+        atlas, 
+        name, 
+        output_dir, 
+        registration_args,
+        transport_args, 
+        shoot_args, 
+        main_reg_dir=None):
+    # estimation of time deformation
+    time_reg_dir = output_dir / name / f'time_reg_{name}'
+    time_reg_dir.mkdir(parents=True, exist_ok=True)
+
+    registration.estimate_registration(
+        source, target, time_reg_dir, **registration_args)
+
+    if main_reg_dir is None:
+        # estimation of subject-to-patient deformation
+        main_reg_dir = output_dir / name / f'main_reg_{name}'
+        main_reg_dir.mkdir(parents=True, exist_ok=True)
+
+        registration.estimate_registration(
+            atlas, source, main_reg_dir, **registration_args)
+
+    # parallel transport of time deformation along subject-to-patient deformation
+    momenta_to_transport = (time_reg_dir / strings.momenta_str).as_posix()
+    control_points_to_transport = (time_reg_dir / strings.cp_str).as_posix()
+    control_points = (main_reg_dir / strings.cp_str).as_posix()
+    momenta = (main_reg_dir / strings.cp_str).as_posix()
+
+    transport_dir = output_dir / name / 'transport'
+    transport_dir.mkdir(parents=True, exist_ok=True)
+    cp, mom = transport(
+        control_points, momenta, control_points_to_transport, momenta_to_transport,
+        transport_dir, **transport_args)
+
+    # Shoot transported momenta from atlas
+    transported_cp = transport_dir / 'final_cp.txt'
+    transported_mom = transport_dir / 'transported_momenta.txt'
+    shoot(
+        control_points=transported_cp.as_posix(),
+        momenta=transported_mom.as_posix(),
+        output_dir=transport_dir, **shoot_args)
+
+    shoot_name = transport_dir / strings.shoot_str.format(transport_args['n_rungs'])
+    subprocess.call(['cp', shoot_name, transport_dir / f'transported_shoot_{name}.vtk'])
```

Comment:
this is very fragile. use `shutil` instead?

but do we really want to copy files?

Reply:
it seems just that Nicolas was not happy with the default name from `deformetrica`.
I'm just wondering if we should change them or simply keep they're folder structure/namings to avoid making it even more complicated